torch/testing/_asserts.py

```diff
@@ -273,7 +284,13 @@ def _equalize_attributes(actual: Tensor, expected: Tensor) -> Tuple[Tensor, Tens
         actual = actual.to(dtype)
         expected = expected.to(dtype)
 
-    if actual.is_sparse and actual.is_coalesced() != expected.is_coalesced():
+    if actual.layout != expected.layout:
+        # These checks are needed, since Tensor.to_dense() fails on tensors that are already strided
```

Comment:
to_dense is also the wrong language because it should be to_strided and then to_strided should probably work like contiguous() and return self if the tensor satisfies the property...

cc @cpuhrsch